public/apps/configuration/utils/tenant-utils.test.tsx

```diff
@@ -0,0 +1,316 @@
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import {
+  GLOBAL_USER_DICT,
+  PRIVATE_USER_DICT,
+  transformTenantData,
+  resolveTenantName,
+  RESOLVED_GLOBAL_TENANT,
+  RESOLVED_PRIVATE_TENANT,
+  globalTenantName,
+  formatTenantName,
+  transformRoleTenantPermissionData,
+  getTenantPermissionType,
+  transformRoleTenantPermissions,
+} from './tenant-utils';
+import {
+  RoleViewTenantInvalidText,
+  TENANT_READ_PERMISSION,
+  TENANT_WRITE_PERMISSION,
+} from '../constants';
+import { TenantPermissionType } from '../types';
+
+describe('Tenant list utils', () => {
+  const expectedGlobalTenantListing = {
+    tenant: GLOBAL_USER_DICT.Label,
+    reserved: true,
+    description: GLOBAL_USER_DICT.Description,
+    tenantValue: GLOBAL_USER_DICT.Value,
+  };
+
+  const expectedPrivateTenantListing = {
+    tenant: PRIVATE_USER_DICT.Label,
+    reserved: true,
+    description: PRIVATE_USER_DICT.Description,
+    tenantValue: PRIVATE_USER_DICT.Value,
+  };
+
+  const expectedTenantListing = {
+    tenant: 'dummy',
+    reserved: false,
+    description: 'dummy',
+    tenantValue: 'dummy',
+  };
+
+  const tenantList = [
+    expectedGlobalTenantListing,
+    expectedPrivateTenantListing,
+    expectedTenantListing,
+  ];
+
+  describe('transform to listing data', () => {
+    const globalTenant = {
+      description: 'Global tenant',
+      reserved: true,
+      hidden: false,
+      static: false,
+      tenant: '',
+      tenantValue: '',
+    };
+
+    const sampleTenant1 = {
+      description: 'dummy',
+      reserved: false,
+      hidden: false,
+      static: false,
+      tenant: '',
+      tenantValue: '',
+    };
+
+    it('transform global tenant', () => {
+      const result = transformTenantData({ global_tenant: globalTenant }, false);
+      expect(result.length).toBe(1);
+      expect(result[0].tenant).toBe(GLOBAL_USER_DICT.Label);
+      expect(result[0].tenantValue).toBe(GLOBAL_USER_DICT.Value);
+      expect(result[0].description).toBe(GLOBAL_USER_DICT.Description);
+      expect(result[0]).toEqual(expectedGlobalTenantListing);
+    });
+
+    it('transform private tenant', () => {
+      const result = transformTenantData({}, true);
+      expect(result.length).toBe(1);
+      expect(result[0].tenant).toBe(PRIVATE_USER_DICT.Label);
+      expect(result[0].tenantValue).toBe(PRIVATE_USER_DICT.Value);
+      expect(result[0].description).toBe(PRIVATE_USER_DICT.Description);
+      expect(result[0]).toEqual(expectedPrivateTenantListing);
```

Comment:
Does the last line cover the first 3 lines?

Reply:
The comment applying to following code.

Reply:
Fixed.